middleware/src/middleware.go

```diff
@@ -1154,6 +1136,51 @@ func (middleware *Middleware) GetServiceInfo() rpcmessages.GetServiceInfoRespons
 	return middleware.serviceInfo
 }
 
+// GetServiceStatus returns the most recent status information of the base and a few of its services
+func (middleware *Middleware) GetServiceStatus() rpcmessages.GetServiceStatusResponse {
+	hostname, err := middleware.redisClient.GetString(redis.BaseHostname)
+	if err != nil {
+		log.Printf("Error getting hostname information. Error: %s", err.Error())
+		errResponse := middleware.redisClient.ConvertErrorToErrorResponse(err)
+		return rpcmessages.GetServiceStatusResponse{ErrorResponse: &errResponse}
+	}
+	isTorEnabled, err := middleware.redisClient.GetBool(redis.TorEnabled)
+	if err != nil {
+		log.Printf("Error getting isTorEnabled information. Error: %s", err.Error())
+		errResponse := middleware.redisClient.ConvertErrorToErrorResponse(err)
+		return rpcmessages.GetServiceStatusResponse{ErrorResponse: &errResponse}
+	}
+	isBitcoindListening, err := middleware.redisClient.GetBool(redis.BitcoindListen)
+	if err != nil {
+		log.Printf("Error getting isBitcoindListening information. Error: %s", err.Error())
+		errResponse := middleware.redisClient.ConvertErrorToErrorResponse(err)
+		return rpcmessages.GetServiceStatusResponse{ErrorResponse: &errResponse}
+	}
+
+	lightningdActive, err := middleware.checkSystemdServiceStatus("lightningd")
+	if err != nil {
+		// only log, since information can still be relayed
+		log.Printf("Error getting lightingd active information from systemctl. Error: %s", err.Error())
+	}
+	electrsActive, err := middleware.checkSystemdServiceStatus("electrs")
+	if err != nil {
+		// only log, since information can still be relayed
+		log.Printf("Error getting electrs active information from systemctl. Error: %s", err.Error())
+	}
+
+	return rpcmessages.GetServiceStatusResponse{
+		ErrorResponse: &rpcmessages.ErrorResponse{
+			Success: true,
+		},
+		Hostname:            hostname,
+		Status:              "-PLACEHOLDER-", // FIXME: This is a placeholder.
+		IsTorEnabled:        isTorEnabled,
+		IsBitcoindListening: isBitcoindListening,
```

Comment:
`IsBitcoindListening` is actually just a setting from redis and doesn't reflect the real-time status of bitcoind. 
Let's also add a `BitcoindStatus` fetched with `checkSystemdServiceStatus`

Reply:
Done.